Create a class to represent the Tree Graph (Issue #1)

imane-aanna opened the pull request.

imane-aanna committed:
Create a class to represent the Tree Graph (Issue #1)

pradal commented on Tree.cs:
```diff
@@ -0,0 +1,91 @@
+namespace csMTG
+{
+   public class Tree
+    {
```
Add documentation for the Tree.
Create a ITree class and inherit from it.

pradal commented on Tree.cs:
```diff
@@ -0,0 +1,91 @@
+        void AddChild(int parentId) {
+
+            if (!(parent.ContainsKey(parentId)))
+                Console.WriteLine("This parent does not exist!");
```
Question: How to manage logging, errors or assert?

pradal commented on Tree.cs:
```diff
@@ -0,0 +1,91 @@
+        // Define the attributes
+        int id { get; set; } = 0;
+        //The key is the parent's id, and the values are a list of children
+        protected Dictionary<int, List<int>> parent = new Dictionary<int, List<int>>();
```
parent need to be replaced by children (parentId -> [chidrenId])

Create a parent attribute:
 protected Dictionary<int, int> parent = new Dictionary<int, int>();

pradal commented:
Excellent.
Add children and parent attributes and a root attribute.
Add little bit more of doc.

imane-aanna committed:
Fixes Issue #6

- Added a root, children and parent attributes.
- Modified the constructor so as to have a parent for root with the id -1 (To signify that a root has no parent).

imane-aanna committed:
Fixes #7

Added a separate function for generating a unique new id

imane-aanna committed:
Fixes #8

Enabled the user to add a child with a specific id through adding a
parameter in the function AddChild. In case the child's id isn't
specified, a unique id will be generated.
Had to cover different scenarios to avoid errors.
Added two functions :
- NewChild which is the quivalent of Python's setDefault .
- ReplaceParent : to change the parent of a child.

imane-aanna committed:
Fixes #3

Write the comments in the conventional format for C#

imane-aanna committed:
Structurate the project

Keep the structure of a Visual Studio project

pradal commented:
First, remove built dirs.
Add them to .gitignore:
See https://github.com/github/gitignore/blob/master/VisualStudio.gitignore

imane-aanna committed:
Remove unneccessary files

Use a .gitignore file in order not to display unneccessary files

imane-aanna committed:
Removed all files that are in the .gitignore

imane-aanna committed:
Rename the folders' names

Renamed the folders' names into: "code" and "test"

imane-aanna committed:
Remove the unneccessary folders again

imane-aanna committed:
Removed all files in .gitignore again

imane-aanna committed:
Fixed the .gitignore file

imane-aanna committed:
Fixes #2 #9

Added tests to evaluate the code
